api/swagger-batch.yaml

```diff
@@ -0,0 +1,310 @@
+    get:
+      tags: [ "Ensembler" ]
+      summary: Returns Ensembler"s details by its ID
+      parameters:
+        - in: path
+          name: project_id
+          schema:
+            $ref: "#/components/schemas/Id"
+          required: true
+        - in: path
+          name: ensembler_id
+          schema:
+            $ref: "#/components/schemas/Id"
+          required: true
+      responses:
+        "200":
+          description: A JSON object representing an instance of the specific type of Ensembler
+          content:
+            application/json:
+              schema:
+                oneOf:
+                  - $ref: "#/components/schemas/PyFuncEnsembler"
+    put:
+      tags: [ "Ensembler" ]
+      summary: Updates existing Ensembler with the data provided in the payload
+      parameters:
+        - in: path
+          name: project_id
+          schema:
+            $ref: "#/components/schemas/Id"
+          required: true
+        - in: path
+          name: ensembler_id
+          schema:
+            $ref: "#/components/schemas/Id"
+          required: true
+      requestBody:
+        description: A JSON object, representing an update to the ensembler
+        required: true
+        content:
+          application/json:
+            schema:
+              oneOf:
+                - $ref: "#/components/schemas/PyFuncEnsembler"
+      responses:
+        "200":
+          description: A JSON object representing an instance of updated Ensembler
+          content:
+            application/json:
+              schema:
+                oneOf:
+                  - $ref: "#/components/schemas/PyFuncEnsembler"
+  "/projects/{project_id}/jobs":
+    get:
+      summary: Returns a list of Ensembling jobs
+      responses:
+        "200":
+          description: TBU
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: "#/components/schemas/EnsemblingJob"
+
+components:
+  schemas:
+    Id:
+      type: "integer"
+      format: "int32"
+    EnsemblerType:
+      type: "string"
+      enum:
+        - pyfunc
+    GenericEnsembler:
+      properties:
+        id:
+          $ref: "#/components/schemas/Id"
+        project_id:
+          $ref: "#/components/schemas/Id"
+        name:
+          type: "string"
+          minLength: 3
+          maxLength: 50
+        type:
+          $ref: "#/components/schemas/EnsemblerType"
+        created_at:
+          type: "string"
+          format: "date-time"
+        updated_at:
+          type: "string"
+          format: "date-time"
+    PyFuncEnsembler:
+      allOf:
+        - $ref: "#/components/schemas/GenericEnsembler"
+        - type: "object"
+          properties:
+            mlflow_url:
+              type: "string"
+              format: "uri"
+            mlflow_experiment_id:
+              $ref: "#/components/schemas/Id"
+            mlflow_run_id:
```

Comment:
Curious, what are the mlflow url/experiement_id and run_id used for this ensembling job? My understanding is that we only require the artifact_uri to build the ensembler and run it.

Reply:
Because Python ensemblers are in fact a specific implementation of MLFlow Python models and are packaged (pickled) and uploaded to the artefact store (GCS) by mlflow client. So this makes it consistent with merlin pyfunc models as well. 
You wouldn't require these fields for the job submission flow